code/modules/mob/mob_movement.dm

```diff
@@ -101,18 +101,21 @@
 	if(Process_Grab())
 		return
 
-	if(mob.buckled) //if we're buckled to something, tell it we moved.
-		return mob.buckled.relaymove(mob, direct)
-
-	if(living_mob && !(living_mob.mobility_flags & MOBILITY_MOVE))
-		return
-
 	if(!mob.lastarea)
 		mob.lastarea = get_area(mob.loc)
 
-	if(isobj(mob.loc) || ismob(mob.loc)) //Inside an object, tell it we moved
-		var/atom/O = mob.loc
-		return O.relaymove(mob, direct)
+	var/mobile = living_mob && (living_mob.mobility_flags & MOBILITY_MOVE)
+
+	if((isobj(mob.loc) || ismob(mob.loc))) //Inside an object, tell it we moved
+		if(mobile)
```

Comment:
```suggestion
	if((isobj(mob.loc) || ismob(mob.loc)) && mobile) //Inside an object, tell it we moved
```
These can probably be smooshed together

Reply:
Actually, I might have to think about this a bit more. I'm not sure if this accomplishes what I want it to, but I'll have to give it another look.